src/TLuaInterpreterMapper.cpp

```diff
@@ -1378,6 +1378,30 @@ int TLuaInterpreter::getAreaRooms(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getAreaRooms1
+int TLuaInterpreter::getAreaRooms1(lua_State* L)
+{
+    const int area = getVerifiedInt(L, __func__, 1, "areaID");
+    const Host& host = getHostFromLua(L);
+    TArea* pA = host.mpMap->mpRoomDB->getArea(area);
+    if (!pA) {
+        lua_pushnil(L);
+        return 1;
+    }
+    lua_newtable(L);
+    QSetIterator<int> itAreaRoom(pA->getAreaRooms());
+    int i = 0;
+    while (itAreaRoom.hasNext()) {
```

Comment:
could we avoid the `while` here? Would this work:

```
   lua_newtable(L);
    int i = 0;
    for (int room : pA->getAreaRooms()) {
        lua_pushnumber(L, ++i);
        lua_pushnumber(L, room);
        lua_settable(L, -3);
    }
```

`while` tends to get into a dangerous habit of eternal loops if not coded properly, so I like to avoid them by default.

Reply:
The `QSetIterator<T>` like all other Qt constant iterators (not the ones with "Mutable" in their names) actually take a copy of the thing they are to iterator over - and they behave differently than a `std::for` type loop. They are NOT going to encounter the sort of problems you describe of "eternal" loops - **provided that you remember to include the `Q<Container>Iterator<T>::next()` inside the loop to do the loop advance**!

https://wiki.qt.io/Iterators

Reply:
That wiki is from 2017 and is unfortunately outdated. In modern C++, `    for (int room : qAsConst(pA->getAreaRooms())) {`  will work OK, see the [qAsConst()](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) macro.

Reply:
Well, yeah! The principles have not changed though - and "if it ain't broke, don't try fixing it"?

!["Yours is the worst code I have ever seen." "Yeah, but at least it runs!"](https://i.redd.it/jnujpmmyzub31.jpg)